docs/features/software-catalog/external-integrations.md

```diff
@@ -1031,3 +1031,360 @@ backend.add(catalogModuleCustomDataParser);
+
+Once the source has no more results, Incremental Entity Provider compares all entities annotated with `@backstage/incremental-entity-provider: <entity-provider-id>` against all marked entities to determine which entities committed by same entity provider were not marked during the last ingestion cycle. All unmarked entities are deleted at the end of the cycle. The Incremental Entity Provider rests for a fixed interval before restarting the ingestion process.
+
+![Diagram of execution of an Incremental Entity Provider](../../assets/software-catalog/software-catalog-incremental-provider-diagram.png)
+
+This approach has the following benefits,
+
+1. Reduced ingestion latency - each burst commits entities which are processed before the entire list is processed.
+2. Stable pressure - each period between bursts provides an opportunity for the processing pipeline to settle without overwhelming the pipeline with a large number of unprocessed entities.
+3. Built-in retry / back-off - Failed bursts are automatically retried with a built-in back-off interval providing an opportunity for the data source to reset its rate limits before retrying the burst.
+4. Prevents orphan entities - Deleted entities are removed as with `full` mutation with a low memory footprint.
+
+### Requirements
+
+The Incremental Entity Provider backend is designed for data sources that provide paginated results. Each burst attempts to handle one or more pages of the query. The plugin will attempt to fetch as many pages as it can within a configurable burst length. At every iteration, it expects to receive the next cursor that will be used to query in the next iteration. Each iteration may happen on a different replica. This has several consequences:
+
+1. The cursor must be serializable to JSON (not an issue for most RESTful or GraphQL based APIs).
+2. The client must be stateless - a client is created from scratch for each iteration to allow distributing processing over multiple replicas.
+3. There must be sufficient storage in Postgres to handle the additional data. (Presumably, this is also true of sqlite, but it has only been tested with Postgres.)
+
+### Installation
+
+1. Install `@backstage/plugin-catalog-backend-module-incremental-ingestion` with `yarn --cwd packages/backend add @backstage/plugin-catalog-backend-module-incremental-ingestion` from the Backstage root directory.
+
+2. Add the following code to the `packages/backend/src/index.ts` file:
+
+```ts title="packages/backend/src/index.ts"
+/* highlight-add-next-line */
+import { catalogModuleCustomIncrementalIngestionProvider } from './extensions/catalogCustomIncrementalIngestion';
+
+const backend = createBackend();
+
+/* highlight-add-start */
+backend.add(
+  import(
+    '@backstage/plugin-catalog-backend-module-incremental-ingestion/alpha'
+  ),
+);
+/* highlight-add-end */
+
+// We have created this in section **Adding an Incremental Entity Provider to the catalog**
```

Comment:
```suggestion
// See the section below for how this provider is created.
```

a bit simpler.

There's always a risk of confusion when you tell people to write an essentially broken piece of code that you fix up later on. Might consider putting the module creation code _first_ in this document, and then finally have a small segment at the bottom of how it gets added to the index.ts.